docs/roles/integrator/integrating.md

```diff
@@ -0,0 +1,372 @@
+  This is generally using wallet-specific protocols. The API used by NEAR Wallet is being defined in [NEP](https://github.com/nearprotocol/NEPs/pull/10).
+
+- **Wallet decodes [Borsh](http://borsh.io/)-serialized transactions and displays approval UI**  \
+  Schema defined [here](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/transaction.ts#L119)
+
+- **SHA-256 hash of transaction signed using ed25519**  \
+  For examples see [nearlib](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/transaction.ts#L198), [wallet-core](https://github.com/trustwallet/wallet-core/blob/951e73abfa0362b4d61202bac4e399a4faae97a8/src/NEAR/Signer.cpp#L20), [Ledger app](https://github.com/nearprotocol/near-ledger-app/blob/5abe5f5d57dff9cefe4535057d7a39f476d32d77/workdir/near-ledger-app/src/crypto/near.c#L7).
+
+- **Transaction sent to NEAR node**  \
+  This is a relatively simple [JSON-RPC call.](https://github.com/nearprotocol/nearlib/blob/8f5063bfee4ea7e7eba1f8dbfc20862534c0febf/src.ts/providers/json-rpc-provider.ts#L41)
+
+- **The App is notified**  \
+  With NEAR Wallet this is a simple redirect to the URL provided by the app when sending transaction.
+
+- **Transaction is available in block explorer**  \
+  Every transaction eventually will show up in [NEAR Explorer](http://explorer.nearprotocol.com) which can be also used by a wallet or exchange to query activity for given account.
+
+
+For additional detail around the life of a transaction on the NEAR platform please see this [Key Concepts: Transaction](/docs/concepts/transaction) page
+
+### Creating transaction offline
+
+Offline transaction signing in 3 steps:
+
+(1) Fetch latest block hash (online: requires access to pull data from the live network)
+
+```js
+const networkStatus = await near.connection.provider.status();
+const blockHash = networkStatus.sync_info.latest_block_hash;
+```
+
+(2) Create transaction (offline)
+
+```js
+const transaction = nearlib.transactions.createTransaction(fromAccount, publicKey, receiverAccount, nonce_for_publicKey, actions, blockHash);
+const bytes = transaction.encode();
+```
+
+(3) Sign transaction (offline with access to the key)
+
+```js
+// WARNING: this sample won't work because Signature is not exported by nearlib
+const signature = await signer.signMessage(message, accountId, networkId);
+const signedTx = new nearlib.transactions.SignedTransaction({transaction, signature: new Signature(signature.signature) });
+```
+
+See other examples of using our [JavaScript SDK here](/docs/roles/developer/examples/nearlib/examples)
+
+
+### Submitting transaction
+
+```js
+let receipt = await near.connection.provider.sendTransaction(signedTx);
+```
+
+### Full working example
+
+You can use a local client-side [playground](/docs/roles/developer/examples/nearlib/guides#prepare-your-playground) to run the code below in your browser.
+
+```js
+ // ADD YOUR ACCOUNT HERE with a valid private key
```

Comment:
probably good to `const nearlib = require('nearlib');`

Reply:
will add the boilerplate imports and connection setup included in the playground to make this clearer

Reply:
⏳